src/study.js

```diff
@@ -37,13 +71,30 @@ export default async function runStudy(devMode) {
           if (newState === runStates.RUNNING) {
           // if the study is running but wasn't previously, let's re-initiate the onPageData listener.
           console.debug("~~~ RS01 running ~~~");
+          Glean.setUploadEnabled(true);
           collectEventDataAndSubmit(rally, devMode);
           } else {
           console.debug("~~~ RS01 not running ~~~");
           // stop the measurement here.
           stopMeasurement();
+          Glean.setUploadEnabled(false);
           }
-      })
+      });
+
+      // If we got to this poin, then Rally is properly
+      // initialized and we can flip collection on.
+      Glean.initialize("rally-zero-one", true, {
```

Comment:
Won't this override the `Glean.setUploadEnabled(false);` on line 80? Or is that just a callback for a possible future event?

Reply:
It shouldn't, that's a callback for a possible future event generated by rally.js :-)